src/ServiceControl/SagaAudit/SagaAuditing.cs

```diff
@@ -35,11 +35,26 @@ public override void Enrich(IReadOnlyDictionary<string, string> headers, IDictio
 
                         foreach (var part in multiSagaChanges.Select(s => s.Split(':')))
                         {
-                            sagasChanges.Add(part[0], part[1]);
+                            var id = part[0];
+                            var thisChange = part[1];
+                            string previousChange;
+                            if (!sagasChanges.TryGetValue(id, out previousChange))
+                            {
+                                sagasChanges[id] = thisChange;
+                            }
+                            else //Completed overrides New overrides Updated
+                            {
+                                if (string.Equals(thisChange, "Completed", StringComparison.OrdinalIgnoreCase)
```

Comment:
👍 for switching to case-sensitive comparison. Not sure I like the "automatic" ordering/comparisons. They look less explicit.